test/blackbox-tests/test-cases/env/env-bins/run.t

```diff
@@ -1,22 +1,26 @@
+  $ CONTEXT=${OPAMSWITCH:-default}
```

Comment:
To be slightly more agnostic of the environment, we could use `$INSIDE_DUNE` which contains the path to the build directory.

Reply:
Hmm it's a bit tricky because `INSIDE_DUNE` is absolute, and the `Path_lexer` everything before `_build/`. One thing we  instead of using `OPAMSWITCH` to is rewrite `/context/` and `/install/context` within the lexer directly. Would this be better?

Reply:
Indeed. In fact, taking a step I just realised that the list of paths printed includes both paths from the inner and outter build directories. This means that if the outer build context is not default, then I'd expect this test to print:

```
- _build/CONTEXT/test/...
- _build/CONTEXT/test/...
- _build/CONTEXT/test/...
- _build/default/.bin
- _build/install/default/bin
```

rather than use `CONTEXT` consistently. So the problem seems to be still there?

Looking at `Path_lexer`, it seems that we try to exclude paths that are not part of the test itself, so in a way it's odd that we include paths coming from the outer dune. What about we change `Path_lexer` to not drop the part before `_build/` and instead filter out any paths that doesn't start with `$PWD`?

Reply:
> rather than use CONTEXT consistently. So the problem seems to be still there?

I think that the way it works, `make all-supported-ocaml-version` sets of workspace (so, contexts) for just the outside build, so it's expected that the inner one is always using the default one. But yes, it's a bit by chance that it works (build path prefix map only replaces the first occurence).

I'll try to implement your suggestion.

Reply:
Done. It's not possible to use `$PWD` directly because it is sometimes outside of `$TESTCASE_ROOT`. So instead I'm just keeping paths that contain 2 `"_build"` segments and there's no need for a lexer at all.

Reply:
Indeed, that's a neat idea!